libraries/AP_Proximity/AP_Proximity_RPLidarA2.cpp

```diff
@@ -97,271 +103,302 @@ void AP_Proximity_RPLidarA2::update(void)
 // get maximum distance (in meters) of sensor
 float AP_Proximity_RPLidarA2::distance_max() const
 {
-    return 16.0f;  //16m max range RPLIDAR2, if you want to support the 8m version this is the only line to change
+    switch (model) {
+    case Model::UNKNOWN:
+        return 0.0f;
+    case Model::A1:
+        return 8.0f;
+    case Model::A2:
```

Comment:
There are 8m and 16m A2 models, the comment should be kept

Reply:
Are there really 8m ranges out there for the A2?